File: Streamish/Models/Video.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Streamish.Models
{
    public class Video
    {
        public int Id { get; set; }

        [Required]
        public string Title { get; set; }

        public string Description { get; set; }

        [Required]
        public string Url { get; set; }

        public DateTime DateCreated { get; set; }

        public int UserProfileId { get; set; }

        public UserProfile UserProfile { get; set; }

        public List<Comment> Comments { get; set; }
    }
}

File: client/src/components/Video.js
import React from "react";
import { Card, CardBody } from "reactstrap";

const Video = ({ video }) => {
  return (
    <Card >
      <p className="text-left px-2"><strong>Posted by:</strong> {video.userProfile.name}</p>
      <CardBody>
        <iframe className="video"
          src={video.url}
          title="YouTube video player"
          frameBorder="0"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen />

        <p><strong>Video Title:</strong> {video.title}</p>
        <p><strong>Description:</strong> {video.description}</p>
        <strong>Comments:</strong>
        {/* {video.comments.map(comment => {
             return <p>Commented by {comment.userProfile.name}: {comment.message}</p>
        })} */}
        {
          video.comments.length !== 0 ?
            video.comments.map(comment => {
                return <p>Commented by {comment.userProfile.name}: {comment.message}</p>}) 
            : 
                <p>N/A</p>
        }


      </CardBody>
    </Card>
  );
};

export default Video;
